### PR TITLE
[release/2.1] Merge master into release/2.1

### PR DESCRIPTION
**Pull request by weshaggard**
Reference release/2.1 branch from master for source-build effort.

We minimally need the partial facade fix https://github.com/dotnet/buildtools/commit/24ef27105b36fe11ab6ac1583867b2a3c28c6742 to address source-build issue in corefx. 

cc @dagood

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets [hunk 1]
```diff
@@ -264,8 +264,8 @@
       Condition="'$(SkipVersionGeneration)' != 'true' AND '$(VersionPropsImported)' != 'true'">
 
     <!-- ############################### -->
-    <Exec Command="git rev-parse --abbrev-ref HEAD 2>&amp;1" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true" Condition="'$(GitBranchName)' == ''">
-      <Output TaskParameter="ConsoleOutput" PropertyName="GitBranchName" />
+    <Exec Command="git describe --all HEAD 2>&amp;1" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true" Condition="'$(RawGitBranchName)' == ''">
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets]: We'll have to use this new variable name in the source-build workaround: [repos/dir.targets#L7-L8](https://github.com/dotnet/source-build/blob/b8d175fa98bb4b52f61a212ee99bfec9327d828b/repos/dir.targets#L7-L8). (BuildTools-side fix tracked by https://github.com/dotnet/buildtools/issues/1980.)

File: src/packages.builds [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
     <!-- Use the same package version for all output packages. -->
     <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
-    <PackageVersion>2.1.0-preview2-$(BuildNumberMajor)-$(BuildNumberMinor)</PackageVersion>
+    <PackageVersion>2.1.0-preview3-$(BuildNumberMajor)-$(BuildNumberMinor)</PackageVersion>
```
Comment (dagood) [inline src/packages.builds]: Would be nice to wait to merge `master` into `release/2.1` again until we start doing preview3 so this change works naturally. For now I guess this has to be reverted.

Comment (weshaggard) [reply src/packages.builds]: We plan to do a full branch merge in the core repo's on Friday but still for preview2 so I will revert this one change.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets (hunk 1 above)
Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets]: Thanks for the pointer. Should we just rename the property back to the old name in BuildTools?

Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets]: Nevermind I think I will just need to update the workaround in source-build.

Comment (dagood) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets]: Sounds good, I think if you're going to make BuildTools-side changes, might as well fix https://github.com/dotnet/buildtools/issues/1980 fully. (Shouldn't be a very complicated change.)

**Pull request merged**